[garylow2001] iP

garylow2001 opened the pull request:
# Chewy, your personalized chatbot assistant

> RRRRAAAWRRRRR - Chewbacca

### Why Chewy?

- track your tasks from the command-line
- datetime aware deadlines and events for added reliability
- ~easy~ **Very easy** to use

### How to run?

1. download it [here](https://github.com/garylow2001/ip/releases/tag/A-Jar)
2. run the duke.jar
3. let Chewy do the rest for you!😉

Alternatively run `main` in Duke.java
```java
public static void main(String[] args) {
        new Duke(DATA_FILE_PATH).run();
    }
```

### Features

- [ ] Manage Todos
- [ ] Manage deadlines
- [ ] Manage Events
- [ ] Find task by keyword
- [ ] Data automatically backed up

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

garylow2001 committed:
Rename, Greet, Exit

garylow2001 committed:
Rename, Greet, Exit.

garylow2001 committed:
minor changes to the text

garylow2001 committed:
Added scanner to read input

garylow2001 committed:
Added tasks list

garylow2001 committed:
Added marking and unmarking tasks, with error handling

garylow2001 committed:
Added Tasks, mark and unmark tasks, and error handling

garylow2001 committed:
Added Deadline, Event and Todo. Handle specific commands such as "/by" and "/from" from each class.

garylow2001 committed:
"Show number of tasks after adding each task"

garylow2001 committed:
Added test cases and expected results

garylow2001 committed:
Added more tests

garylow2001 committed:
Added Duke Exceptions and a help message

garylow2001 committed:
Change placement of throw and catch

garylow2001 committed:
Added delete command. Added test cases. Added NoTaskFoundException.

garylow2001 committed:
Add save tasks and load tasks function

garylow2001 committed:
Merge branch 'branch-Level-7'

garylow2001 committed:
Add Dates and Times

garylow2001 committed:
Merge branch 'branch-Level-8'

garylow2001 committed:
Refactor code into Parser, Storage, TaskList, Ui

garylow2001 committed:
Merge branch 'A-MoreOOP'

garylow2001 committed:
Refactor Exceptions and Tasks into packages

garylow2001 committed:
Merge branch 'A-Packages'

garylow2001 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

garylow2001 committed:
Add JUnit testing

garylow2001 committed:
Merge branch 'branch-A-JUnit'

garylow2001 committed:
Add JAR file

garylow2001 committed:
Add JavaDocs to methods and classes

garylow2001 committed:
Refactor Duke, Storage, Task, Event and Todo to eliminate magic literals

garylow2001 committed:
Add find operation to find task by keyword

garylow2001 committed:
Add JavaDocs to new methods created from find operation

garylow2001 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Tasks/Task.java
#	src/main/java/Tasks/Todo.java

garylow2001 committed:
Merge branch 'branch-Level-9'

SelwynAng commented on src/main/java/Tasks/Event.java:
```diff
@@ -0,0 +1,83 @@
+public class Event extends Task {
+    private LocalDateTime startTime;
+    private LocalDateTime endTime;
+    public Event(String description) throws DukeException {
```
Perhaps adding Javadocs for your methods will help readers be able to better understand the functionality of your methods. Do consider adding it!

SelwynAng commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,114 @@
+        String taskType = lineParts[0];
+        boolean isDone = lineParts[1].equals("X");
+        String description = lineParts[2];
+        switch (taskType) {
```
I think that you should not indent your case statements. According to the coding standards, case statements should be on the same level of indentation as the switch statement.

SelwynAng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,64 @@
+            String userInput = scanner.nextLine();
+
+            try {
+                if (userInput.equals("bye")) {
```
Perhaps try refactoring this into a switch statement which may result in a cleaner-looking code. Good to standardise whether you want to use if-else or switch statements throughout your code!

SelwynAng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,64 @@
+import java.util.Scanner;
 public class Duke {
+    private final Ui _ui;
+    private TaskList _taskList;
```
Perhaps you should name your variable as taskList instead of _taskList. I think it's better not to let a variable name start with a hyphen. It's good practice to use camelCase for your variable names.

SelwynAng commented on src/main/java/Tasks/Event.java:
```diff
@@ -0,0 +1,83 @@
+        }
+    }
+
+    public Event(String description, String startTimeString, String endTimeString) {
```
Would be nice to add Javadocs for all of your methods to better explain the functionality of your methods!

garylow2001 committed:
Add JavaFX and FXML files for UI

garylow2001 committed:
Refactor TaskList, Ui and Duke to return Duke response to user input

garylow2001 committed:
Merge branch 'branch-Level-10'

SelwynAng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,78 @@
+     */
+    public String getResponse(String userInput) {
+        try {
+            if (userInput.equals("bye")) {
```
Maybe it would be better to use switch statements here instead of verbose if else statements?

SelwynAng commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,60 @@
+     * @return The welcome message.
+     */
+    public String showWelcome() {
+        String logo = "I'm Chewy,\n" +
```
Perhaps the logo string does not need a "+" operator, it can just be condensed into 1 string.

garylow2001 committed:
Add Checkstyle

garylow2001 committed:
Configure build.gradle to launch new jar file

garylow2001 committed:
Add assertions to DialogBox, Parser, TaskList and Tasks

DialogBox: ensure text is not empty
Parser: ensure userInput is not empty
TaskList: ensure pre-conditions such as userInput and taskId are of correct values,
ensure post-conditions such as selectedTask (mark and unmark) change correctly
Deadline, Event, Task: ensure input string parse correctly in constructor

The classes do not check internal invariants, pre-conditions and post-conditions for correct values

Let's add assertions to check for that these values are correct.

garylow2001 committed:
Refactor files into packages and code for better code quality

Structure of the project is not organized with many files in the "java" folder.
Let's extract DialogBox and MainWindow into Gui package.
Let's extract all other files in "java" to "java.Chewy" package.

Multiple levels of abstraction in Deadline and Event.
Let's extract out long methods to follow SLAP.

garylow2001 committed:
Merge pull request #1 from garylow2001/branch-A-Assertions

Add assertions to DialogBox, Parser, TaskList and Tasks

garylow2001 committed:
Merge branch 'master' of github.com:garylow2001/ip

garylow2001 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Chewy/TaskList.java
#	src/main/java/Tasks/Deadline.java
#	src/main/java/Tasks/Event.java

garylow2001 committed:
Merge pull request #3 from garylow2001/branch-A-CodeQuality

Refactor files into packages and code for better code quality

garylow2001 committed:
Merge branch 'master' of github.com:garylow2001/ip

garylow2001 committed:
Create DoAfter task and TaskSpecificException

garylow2001 committed:
Merge branch 'branch-BCD-Extension'

garylow2001 committed:
Add Ui screenshot

garylow2001 committed:
Add Chewbacca emotes to Ui

garylow2001 committed:
Merge branch 'branch-A-Personality'

garylow2001 committed:
Edit User Guide

garylow2001 committed:
Merge branch 'branch-A-UserGuide'

garylow2001 committed:
Fix checkstyle errors and package naming convention errors

garylow2001 committed:
Update build.gradle to launch jar file
